util/prefix_validate.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"errors"
+	"regexp"
+)
+
+/*
+IsValidTagPrefix runs a series of regex checks to ensure that the tag prefix is valid
+  Tags cannot begin or end with, or contain multiple consecutive / characters.
+  They cannot contain any of the following characters \, ?, ~, ^, :, * , [, @.
+  They cannot contain a space.
+  They cannot end with a . or have two consecutive .. anywhere within them.
+  Tags are not case-sensitive.
```

Comment:
This description does not match exactly what the function does any more. I assume it's better to also describe what is allowed here then to describe what's forbidden.